packages/core/lib/scope.js

```diff
@@ -117,11 +124,29 @@ const resolveInternalScope = (name, scope) => {
   }
 };
 
+/**
+ * Recursively check if our scope meets certain conditions (usually used with isBrowser, isPage, or isRepl)
+ * @param {scope => boolean} f the function used to check if we're in a scope
+ * @param {Scope} scope The scope to use
+ */
+const validateScope = (f, scope, withError = false) => {
+  const inScope = f(scope);
+  if (inScope) return true;
+  if (scope.parent) {
+    return validateScope(f, scope.parent);
```

Comment:
The recursion pattern for other functions in this file uses the base case check for  `scope === null` to throw the error. Can you do that just for consistency, instead of checking `if (scope.parent)` and continuing. I personally just fine it easier to read recursion that way

base case

and then any other pro-order recursion steps, followed by the tail call recursion